packages/features/calendars/DatePicker.tsx

```diff
@@ -122,20 +122,19 @@ const Days = ({
   // Create placeholder elements for empty days in first week
   const weekdayOfFirst = browsingDate.date(1).day();
   const currentDate = minDate.utcOffset(browsingDate.utcOffset());
+
   const availableDates = (includedDates: string[] | undefined) => {
     const dates = [];
     const lastDateOfMonth = browsingDate.date(daysInMonth(browsingDate));
-    for (
-      let date = currentDate;
-      date.isBefore(lastDateOfMonth) || date.isSame(lastDateOfMonth, "day");
-      date = date.add(1, "day")
-    ) {
+    let date = currentDate;
+    do {
+      date = date.date(date.date() + 1);
```

Comment:
In the existing code, the date was updated after dates were pushed to the array. Am I reading this correctly that the first iteration through this do/while will actually update the date before pushing any dates to the array. Will we miss the first day here somehow?